### PR TITLE
Upload file to s3: make public a file on upload and allow to configure destination folder

### PR DESCRIPTION
**Pull request by lautaroalvarez**
### Component/Part
s3 imageRouter (`lib/web/imageRouter/s3.js`)

### Description
New configurations:
- s3folder: (string) folder to save the files inside bucket
- s3publicFiles: (boolean) indicate if should send ACL parameters

When `s3publicFiles` is true we add the ACL parameter to the `s3.putObject` function.

Aws doc: https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/S3.html#putObject-property . The paremeter is `ACL`. The value we want is "public-read"

### Steps
- [x] Added implementation
- [x] Added / updated documentation
- [x] I read the [contribution documentation](https://github.com/hedgedoc/hedgedoc/blob/develop/CONTRIBUTING.md) and
  made sure that:
  - My commits are signed-off to accept the DCO.
  - This PR targets the correct branch: `master` for 1.x & docs, `develop` for 2.x

### Related Issue(s)
https://github.com/hedgedoc/hedgedoc/issues/2384

### CODE REVIEW

**Comment by davidmehren** [comment]
We do not accept new features for HedgeDoc 1.x anymore, as stated in the [README](https://github.com/hedgedoc/hedgedoc#state-of-the-project):
> The 1.x release is maintenance-only. We do not accept feature requests or PRs for this release anymore and may choose to close non-critical bug reports, if the bug will be non-existent in 2.0.

**Comment by lautaroalvarez** [comment]
@davidmehren but It's a bug: I can't use it with my digitalocean space as S3 because it uploads files without public permissions. The important change is the line https://github.com/hedgedoc/hedgedoc/pull/2385/files#diff-49ad4b49eb8a16cecc65109370e1545513dc4017586b811f04dfc4124a449f3eR37 . Maybe I have to remove the other changes because they can be interpreted as "new features".

I have added a new environment configuration "s3publicFiles" because I don't want to change the current use of s3.

Is there any way I can apply this fix? Because I really want to use Hedgedoc with my digitalocean space.

**Comment by davidmehren** [comment]
We just discussed this again and decided to accept this as a bugfix.

File: docs/content/configuration.md [hunk 1]
```diff
@@ -249,11 +249,12 @@ you don't have to use either of these.
...
+|             | `CMD_S3_REGION`            | **no default**, `ap-northeast-1`                                                                                  | AWS S3 region                                                                                                    |
+| `s3bucket`  | `CMD_S3_BUCKET`            | **no default**                                                                                                    | AWS S3 bucket name                                                                                                      |
 |             | `CMD_S3_ENDPOINT ENV`      | **no default**                                                                                                    | S3 API endpoint if you don't use AWS name                                                                                                  |
+| `s3folder`  | `CMD_S3_FOLDER`            | 'uploads'                                                                                                         | Folder where the files will be saved inside the bucket                                                                                   |
```
Comment (davidmehren) [inline docs/content/configuration.md:257]: The `s3publicFiles` option should also be documented.

File: docs/content/configuration.md [hunk 2]
```diff
@@ -246,14 +246,16 @@ you don't have to use either of these.
...
+|                 | `CMD_S3_ENDPOINT ENV`      | **no default**                                                                                                    | S3 API endpoint if you don't use AWS name                                                                                                  |
+| `s3folder`      | `CMD_S3_FOLDER`            | 'uploads'                                                                                                         | Folder where the files will be saved inside the bucket                                                                                     |
+| `s3publicFiles` | `CMD_S3_PUBLIC_FILES`      | `false`                                                                                                           | If se to `true` set public permissions to files when uploading to s3 storage                                                               |
```
Comment (davidmehren) [inline docs/content/configuration.md:258]:
```suggestion
| `s3publicFiles` | `CMD_S3_PUBLIC_FILES`      | `false`                                                                                                           | If set to `true`, a `public-read` ACL is applied to uploaded objects                                                               |
```

**Review by davidmehren — CHANGES_REQUESTED** [review]
Just one last thing regarding wording

File: docs/content/configuration.md (hunk 2 above)
Comment (lautaroalvarez) [reply docs/content/configuration.md]: Much better

**Review by InnayTool — APPROVED** [review]

**Review by davidmehren — APPROVED** [review]

**Pull request merged**